utils/author_catalog.py

```diff
@@ -0,0 +1,66 @@
+import collections
+import csv
+from pathlib import Path
+from typing import Dict
+
+catalog_file_path = str(Path(__file__).parents[1]) + "/resources/" + "pg_catalog.csv"
+
+author_list = (
+    "Jefferson, Thomas, 1743-1826",
+    "Stevenson, Robert Louis, 1850-1894",
+)  # sirasiyla 1, 2, sonuc cikmasi gerekiyor.
+
+
+def read_csv(filepath=catalog_file_path):
+    lines = csv.reader(open(filepath), delimiter=",")
+    next(lines)  # skip the header.
+    for line in lines:
+        # data has some random new lines. Replace them with white space, then strip it.
+        yield list(map(lambda s: s.replace("\n", " ").strip(), line))
+
+
+def parse_data_row(row: list):
+    return (
+        row[0],  # book number
+        row[3],  # title
+        row[4],  # language
+        row[5],  # authors
+    )
+
+
+def list_author_data(row, list_of_author=author_list, language="en"):
+    authors_set = set(
+        list_of_author
+    )  # Q burada tuple'i set'e cevirdim hizlansin diye yoksa gereksiz bir hamle mi?
+    book_number, book_title, book_lang, authors = row[:]
+    if authors in authors_set and book_lang == "en":
+        return row
+
+
+def create_catalog(filepath: str = catalog_file_path) -> Dict:
+    """
+
+    Note:
+        If there are different versions of the same book in the source catalog, it takes only the last version
+    since it saves by title (the last one overwrites the previous version).
+
+
+    Args:
+        filepath (str): The first argument. The filepath of the source catalog of the related database.
+
+    Returns:
+
+        Return a dictionary contains authors' dictionaries that contain book title and book id pairs.
+
+    """
+    author_book_catalog = collections.defaultdict(dict)
+    for row in read_csv(filepath):
+        parsed_row = parse_data_row(row)
+        if list_author_data(parsed_row) is not None:
+            author_book_catalog[parsed_row[3]][parsed_row[1]] = parsed_row[0]
```

Comment:
A few issues here:
- not sure I understand the reason for having two dictionaries. Why do we just have a dictionary of `Author-names` -> `Entry` (read my comment regarding dataclasses and `Entry` above).
- second issue is that you didn't use at all any merits of `defaultdict`. Either don't create a `defaultdict` or use it somehow.
- you didn't use the language. 

what we would like to have is that we want to have a collections of items by an author. I'll suggest you to have a dictionary of single `Author -> [Entry, ...]` map. In order to do that you need to parse authors (is there any delimiter to parse it?) and then append it to the that author's list of entries. Hint: `defaultdict(list)` might be helpful.